crossdock/src/tchannel_server.js

```diff
@@ -46,17 +57,24 @@ export default class TChannelServer {
     }
 
     handleTChannelRequest(context: any, req: any, head: any, body: any, callback: Function) {
-        callback(null, {
+        let startRequest: boolean = false;
+        let traceRequest = body.request;
+        let spanContext = TChannelBridge.getSpanFromTChannelRequest(this._tracer, 'tchannel-server-span', head).context();
```

Comment:
Two problems here
  * the span is, apparently, never finished, thus is a bad example of using the API
  * I would prefer if we defined an API where (a) the user did not have to deal with the span manually, and (b) the user **had** to accept with some "context" object instead

Rather than making the user call getSpanFromTChannelRequest, I think we can create a decorator around the whole request handler (which incidentally already takes a context). Specifically, in L52 you're registering   `handleTChannelRequest` as a handler with tchannel. Instead you should be registering

`TracedHander(handleTChannelRequest)`

where TracedHander would take care of the span etc., and then delegate to the actual handler.